### DISCUSSION TITLE
Please help with my URL, SSL with Apache2 on Linode

### DISCUSSION THREAD

**Discussion by ghost** [owncast/owncast#1824] [Q&A] [answered]
OK so I'm amazed that I got Owncast running on my VPS. 
My Linode is Debian 10 with 2 CPU Cores & 80 GB Storage
CPU: AMD EPYC 7542 32- (2) @ 2.899GHz with 4GB RAM

I have 3 Apache2 virtual hosts on the VPS. All of them have a valid lets-encrypt SSL certificate installed. 
I'm really confused about a few things. 

1.) The logos and custom icons (favicon and my edited owncast logo) do not show up if I go to the URL http://pastlife.works:8080 but when I enter the http://IP:8080 or any of the other virtual hosts,  I can see my custom icons.  Why would the one virtual host that I want to use not display the logos?  This really wouldn't be an issue if I could figure out reverse proxy I think, but IDK about that.

2.) I went over this page: [Apache2 - Owncast](https://owncast.online/docs/sslproxies/apache/)  and I'm not sure what to do with the .config file there.  Do I need to create a new domain and new virtual host with apache and get a new SSL cert for that so I can make the .config file like this guide? I'm really lost...

As I said above, I have 3 virtual hosts, so I have 6 different config files for those in `/etc/apache2/site-enabled/`  (one for the http version and one file for the https config.)

What I want is to be able to go to https://pastlife.works/live/ and be able to see the stream. 
Or it would be a lot better if I can set it up to be at https://live.pastlife.works but I dont know how to set up a sub domain at all. 

Customization and embedding will have to come later because IDK what I'm doing at the moment. I just want the stream to be pointed to my URL, with my custom icons.  

Please help this noob. I have been on Linux for only 2.5 years or so, but I feel that I have learned a lot. I'm still really confused about proxies, reverse proxy, and SSL 

If you care to answer, please also explain why do I need reverse proxy to get the SSL URL for the stream?

Thank you so so much. 
Much love and keep up all the great work you are doing. This is a revolutionary project.

**Comment by ghost** [comment]
Update: 

I figured out the sub domain thing.  I did it with an A Record, as well as a CNAME record. Both ways I tried I was able to get to the Owncast GUI by typing in the URL: http://live.pastlife.works:8080  but this is undesirable since I would like to close port 8080 and reverse proxy to have only port 80 /443 open. Still confused, but at least I got the software running. . 

 I do have my server set up with SSH keys and root login disabled, but I'm wondering, how much of a security risk is keeping the stream the way it is, no SSL/reverse proxy and port 8080 open to public traffic?
 
What if I dont care if people are scared of visiting my stream with a "not secure" text in the address bar?
Is it a risk to my server, besides anyone being able to view the video and plain text from anywhere as it's unencrypted data?

**Comment by gabek** [comment]
Encryption and security aside, there are a couple features that won't be available to you. There might be others, but this is what came to mind.

- Embedding your stream onto another page.
- Support for the [social features](https://owncast.online/docs/social/)

Other than that it's completely your choice if you don't want to use SSL.

**Reply by ghost** [reply]
Thank you for reminding me to not be lazy. It is a very bad thing for me to go half way. Promotes procrastination.  
I definitely want to be able to embed the stream on a page of my website, and taking advantage of the Fediverse social features is a huge advantage. 

I'll be posting on the Apache user forums today to figure out what I need to do to get a reverse proxy working. 
This is like sending a 5 year old to go fishing with just the line. It's possible, but highly unlikely he'll catch a fish.

**Comment by ghost** [comment]
I'm seeking help at Apache and Let's Encrypt Community. 
So far the peeps at L.E. are helping me make some progress I think. 
If anyone wants to help, there may be some useful information here: 
[https://community.letsencrypt.org/t/help-with-apache-reverse-proxy-to-get-ssl-for-my-owncast-stream-server/175942/2](https://community.letsencrypt.org/t/help-with-apache-reverse-proxy-to-get-ssl-for-my-owncast-stream-server/175942/2)

If anyone can help solve this I can donate a few bucks worth of XMR. 
I dont have much but I can at least buy you a beer or coffee.

Much thanks to all the Owncast people.

**Comment by ghost** [comment, marked as answer]
It's fixed.  I had to run this command for it to work: `sudo a2enmod headers` 
The steps taken were as follows: 

1. create an HTTP site for "live.example.com" in `/etc/apache2/sites-available/`
2. get a certbot certificate for "live.example.com"
3. create an HTTPS websitesite for "live.example.com" [simple "hello world" page] - may not be needed
4. proxy https://live.example.com to local system and port (see below for config)

**Here is the way I configured the le-ssl.conf for apache:** 

```
 <IfModule mod_ssl.c>
<VirtualHost *:443>

        ServerName live.pastlife.works
        ServerAdmin email
        ServerAlias live.pastlife.works
        DocumentRoot /var/www/live.pastlife.works

        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined

        ProxyRequests       Off
        ProxyPreserveHost   On
        AllowEncodedSlashes NoDecode

        <Proxy *>
          Order deny,allow
          Allow from all
        </Proxy>
## order matters here, RequestHeader Flags before ProxyPass flags ##
        RequestHeader    set X-Forwarded-Proto "https"
        RequestHeader    set X-Forwarded-Port "443"

        ProxyPass        / http://localhost:8080/
        ProxyPassReverse / http://localhost:8080/

## Without this section chat does not work ##        
        RewriteEngine On
        RewriteCond %{HTTP:UPGRADE} ^WebSocket$           [NC,OR]
        RewriteCond %{HTTP:CONNECTION} ^Upgrade$          [NC]
        RewriteRule .* ws://127.0.0.1:8080%{REQUEST_URI}  [P,QSA,L]


## order matters here, Include first ##
Include /etc/letsencrypt/options-ssl-apache.conf
SSLCertificateFile /etc/letsencrypt/live/live.pastlife.works/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/live.pastlife.works/privkey.pem

</VirtualHost>
</IfModule>
```
